pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1080,102 +1077,126 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                 }
                 Optional<Map<String, String>> subscriptionProperties = SubscriptionOption.getPropertiesMap(
                         subscribe.getSubscriptionPropertiesList());
-                service.getTopic(topicName.toString(), createTopicIfDoesNotExist)
-                        .thenCompose(optTopic -> {
-                            if (!optTopic.isPresent()) {
-                                return FutureUtil
-                                        .failedFuture(new TopicNotFoundException(
-                                                "Topic " + topicName + " does not exist"));
-                            }
 
-                            Topic topic = optTopic.get();
+                CompletableFuture<Boolean> isAuthorizedToCreateTopicFuture = isNamespaceOperationAllowed(
+                        topicName.getNamespaceObject(),
+                        NamespaceOperation.CREATE_TOPIC
```

Comment:
Maybe we can add this before [line 1026](https://github.com/apache/pulsar/pull/17411/files#diff-1e0e8195fb5ec5a6d79acbc7d859c025a9b711f94e6ab37c94439e99b3202e84L1026), and use this method `isAuthorizedFuture.thenCombine(isAuthorizedToCreateTopicFuture, (isAuthorized, isAuthorizedToCreateTopic)`.

Reply:
This will prevent produce/consume of topic if it already exist but create_topic operation not authorized, nope?

Reply:
Oh nvm, got it.

Reply:
@gaoran10 done.